Library/Homebrew/version.rb

```diff
@@ -379,3 +392,29 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
   end
 end
+
+class HeadVersion < Version
+  attr_reader :commit
+
+  def initialize(val)
+    if val.respond_to?(:to_str)
+      @version = val.to_str
+    else
+      raise TypeError, "Version value must be a string; got a #{val.class} (#{val})"
+    end
+    @commit = @version[/HEAD-(.+)/, 1]
```

Comment:
Regular expressions are greedy by default, thus I can't think of a situation where this wouldn't already do the right thing. Nonetheless I'd prefer to be more explicit: `/^HEAD-(.+)$/`.